rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -39,6 +39,20 @@
 
 namespace
 {
+/**
+ * Trivial std::unique_lock wrapper providing constructor that allows Clang Thread Safety Analysis.
+ * The std::unique_lock does not have these annotations.
+ */
+class RCPPUTILS_TSA_SCOPED_CAPABILITY TSAUniqueLock : public std::unique_lock<std::mutex>
```

Comment:
Non-blocking: I wonder if we should add this to `rcpputils`. I don't anticipate clang-libcxx adding it any time soon, so we may want it in a shared place.

Reply:
While agree with you that we need to move this definition in to the rcpputils, I don't have time and resources to do this currently.